typescript-advance/exercise-1.ts

```diff
@@ -0,0 +1,169 @@
+type Listener<EventType> = (ev: EventType) => void;
```

Comment:
Overall it looks great! I had so much fun while testing pokemons :) 
I just wanna left couple of comments:
- let's create separate folder for Interfaces
- let's create separate folder for Types
- let's keep classes and functions in separate file
- let's create the file like test.ts or demo.ts and put all code where we're creating pokemons and calling methods
It will improve code readability & maintainability

Reply:
The codebase structure is improved.